superset/dashboards/dao.py

```diff
@@ -37,40 +36,20 @@
 
 class DashboardDAO(BaseDAO):
     model_cls = Dashboard
-    base_filter = DashboardFilter
+    base_filter = DashboardAccessFilter
 
     @staticmethod
     def get_by_id_or_slug(id_or_slug: str) -> Dashboard:
-        query = (
-            db.session.query(Dashboard)
-            .filter(id_or_slug_filter(id_or_slug))
-            .outerjoin(Slice, Dashboard.slices)
-            .outerjoin(Slice.table)
-            .outerjoin(Dashboard.owners)
-            .outerjoin(Dashboard.roles)
-        )
-        # Apply dashboard base filters
-        query = DashboardFilter("id", SQLAInterface(Dashboard, db.session)).apply(
-            query, None
-        )
-        dashboard = query.one_or_none()
+        dashboard = Dashboard.get(id_or_slug)
         if not dashboard:
             raise DashboardNotFoundError()
+        if is_feature_enabled("DASHBOARD_RBAC"):
```

Comment:
the check for the ff happens also inside the raise_for_dashboard_access so it seems a bit wiered to repeat to check

Reply:
Good point.